### PR TITLE
Fix TFRemBertEncoder all_hidden_states

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Current `TFRemBertEncoder`  adds (initial, after a projection) `hidden_states` twice to `all_hidden_states`.
This PR fixes it.

In the test, it comments `# RemBERT also returns the upprojected word embeddings as an hidden layers`, but the code actually added the projected embedding twice. Furthermore, PyTorch's RemBERT only added the projected one.

@Rocketknight1 @gante

### CODE REVIEW

**Review by gante — APPROVED** [review]
Interesting, the PT implementation does not have this -- and this difference has been present since the first commit of RemBert 🤔 

@sgugger double-checking if I can merge

File: tests/test_modeling_tf_rembert.py [hunk 1]
```diff
@@ -90,8 +90,7 @@ def __init__(
         self.num_choices = 4
         self.scope = None
 
-        # RemBERT also returns the upprojected word embeddings as an hidden layers
-        self.expected_num_hidden_layers = self.num_hidden_layers + 2
+        self.expected_num_hidden_layers = self.num_hidden_layers + 1
```
Comment (sgugger) [inline tests/test_modeling_tf_rembert.py:93]: I think this line can now be removed as it's the default.

**Review by sgugger — APPROVED** [review]
Very nice catch, thanks for fixing!

File: tests/test_modeling_tf_rembert.py (hunk 1 above)
Comment (ydshieh) [reply tests/test_modeling_tf_rembert.py:93]: ok

**Comment by ydshieh** [comment]
Applied suggestion

https://github.com/huggingface/transformers/pull/15510#discussion_r799506824

**Pull request merged**